extern/boostd-data/yugabyte/cassmigrate/20230913144459_deals_add_is_direct.go

```diff
@@ -0,0 +1,48 @@
+package cassmigrate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/yugabyte/gocql"
+	"golang.org/x/sync/errgroup"
+)
+
+// ts20230913144459_dealsAddIsDirectDealColumn adds a new column isDirectDeal to pieceDeal table
+func ts20230913144459_dealsAddIsDirectDealColumn(ctx context.Context, session *gocql.Session) error {
+	qry := `ALTER TABLE PieceDeal ADD IsDirectDeal BOOLEAN`
+	err := session.Query(qry).WithContext(ctx).Exec()
+
+	if err != nil {
+		return fmt.Errorf("creating new column IsDirectDeal: %w", err)
+	}
+
+	qry = `SELECT DealUuid from PieceDeal`
+	iter := session.Query(qry).WithContext(ctx).Iter()
+
+	dealIsDirect := make(map[string]bool)
+	var id string
+	for iter.Scan(&id) {
+		dealIsDirect[id] = false
+	}
+	if err := iter.Close(); err != nil {
+		return fmt.Errorf("getting deal miner address: %w", err)
+	}
+
+	eg, ctx := errgroup.WithContext(ctx)
+	eg.SetLimit(64) // only run 64 go routines at a time
+	for did, isDD := range dealIsDirect {
+		i := did
+		dd := isDD
+		eg.Go(func() error {
+			qry1 := `UPDATE PieceDeal SET IsDirectDeal = ? WHERE DealUuid = ?`
+			err1 := session.Query(qry1, dd, i).WithContext(ctx).Exec()
+			if err1 != nil {
+				return fmt.Errorf("setting deal %s miner address to %t: %w", i, dd, err1)
+			}
+			return nil
+		})
+	}
+
+	return eg.Wait()
```

Comment:
I'm not sure that we need to set IsDirectDeal to false explicitly - IIUC the database with automatically set a new boolean column to false by default.

Reply:
I don't think Cassandra does default values. It will set it to NULL as everything is key:value pair in the lower layers. I will  check with Yugabyte team about this.

Reply:
hmm yeah I think you're right, it will set the column to null.
If that's the case then I think we can still simplify this a bit - we don't need to update all rows one by one, we can just do a single update for all rows:
`UPDATE PieceDeal SET IsDirectDeal = false`

Reply:
Cassandra doesn't allow setting a column value for all rows. I tried different approaches but none of them worked. 

```
  (ql error -12)
go-tests_1  |     svc_test_util.go:348: 
go-tests_1  |           Error Trace:    /go/src/svc/svc_test_util.go:348
go-tests_1  |                                                   /go/src/svc/svc_yugabyte_test.go:52
go-tests_1  |           Error:          Received unexpected error:
go-tests_1  |                           starting local index directory service: running cassandra migrations: running migration ts20230913144459_dealsAddIsDirectDealColumn: setting IsDirectDeal to false: Invalid CQL Statement. Missing partition key
go-tests_1  |                           UPDATE PieceDeal SET IsDirectDeal = FALSE
go-tests_1  |                           ^^^^^^
go-tests_1  |                            (ql error -12)
go-tests_1  |           Test:           TestCleanupYugabyte
```

Reply:
Interesting, I guess that makes sense as it's not really a relational db.
Ok in that case I'll resolve the comment.